File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
import java.io.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> hashMap = new HashMap<String, Integer>();
        
        for(String part : participant) {
            hashMap.put(part, hashMap.getOrDefault(part, 0) + 1);
        }
        
        for(String com : completion) {
            hashMap.put(com, hashMap.get(com) - 1);
        }
        
        for(String hash : hashMap.keySet()) {
            if(hashMap.get(hash) == 1) {
                answer = hash;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    unordered_map<string, int> participant_map;

    for (auto part : participant) {
        participant_map[part]++;
    }

    for (auto comp : completion) {
        participant_map[comp]--;
    }

    for (auto part : participant_map) {
        if (part.second > 0) {
            return part.first;
        }
    }

    return ""; 
}